docs/book/src/developer/architecture/controllers/machine-pool.md

```diff
@@ -0,0 +1,114 @@
+increments the number of ready replicas. When all replicas are ready and the infrastructure ref is also  
+`Ready`, the machine pool controller marks the machine pool as `Running`.
+
+## Contracts
+
+### Cluster API
+
+Cluster associations are made via labels.
+
+#### Expected labels
+
+| what | label | value | meaning |
+| --- | --- | --- | --- |
+| MachinePool | `cluster.x-k8s.io/cluster-name` | `<cluster-name>` | Identify a machine pool as belonging to a cluster with the name `<cluster-name>`|
+
+### Bootstrap provider
+
+The BootstrapConfig object **must** have a `status` object.
+
+To override the bootstrap provider, a user (or external system) can directly set the `MachinePool.Spec.Bootstrap.DataSecretName`
+field. This will mark the machine as ready for bootstrapping and no bootstrap data secret name will be copied from the
+BootstrapConfig object.
+
+#### Required `status` fields
+
+The `status` object **must** have several fields defined:
+
+* `ready` - a boolean field indicating the bootstrap config data is generated and ready for use.
+* `dataSecretName` - a string field referencing the name of the secret that stores the generated bootstrap data.
+
+#### Optional `status` fields
+
+The `status` object **may** define several fields that do not affect functionality if missing:
+
+* `failureReason` - a string field explaining why a fatal error has occurred, if possible.
+* `failureMessage` - a string field that holds the message contained by the error.
+
+Example:
+
+```yaml
+kind: MyBootstrapProviderConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+status:
+    ready: true
+    dataSecretName: "MyBootstrapSecret"
+```
+
+### Infrastructure provider
+
+The InfrastructureMachine object **must** have both `spec` and `status` objects.
+
+#### Required `spec` fields
+
+The `spec` object **must** have at least one field defined:
+
+* `providerIDList` - the list of cloud provider IDs identifying the instances.
+
+#### Required `status` fields
+
+The `status` object **must** have at least one field defined:
+
+* `ready` - a boolean field indicating if the infrastructure is ready to be used or not.
+
+#### Optional `status` fields
+
+The `status` object **may** define several fields that do not affect functionality if missing:
+
+* `failureReason` - is a string that explains why a fatal error has occurred, if possible.
+* `failureMessage` - is a string that holds the message contained by the error.
+
+Example:
+```yaml
+kind: MyMachinePool
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+spec:
+    providerIDList: 
+      - cloud:////my-cloud-provider-id-0
+      - cloud:////my-cloud-provider-id-1
+status:
+    ready: true
+```
+
+### Secrets
+
+The machine pool controller will use a secret in the following format:
+
+| secret name | field name | content |
+|:---:|:---:|---|
+|`<cluster-name>-kubeconfig`|`value`|base64 encoded kubeconfig that is authenticated with the child cluster|
```

Comment:
On line 14 we have _target cluster_,
on line 17,24 we have _workload cluster_.

Suggesting we use _workload cluster_ everywhere since that's what we have in the glossary.

```suggestion
|`<cluster-name>-kubeconfig`|`value`|base64 encoded kubeconfig that is authenticated with the workload cluster|
```